### ISSUE TITLE
Final Peer Review

### ISSUE THREAD

**Issue by mcguirecooper** [zgk2003/covid19_prediction#12]
The object of this project is to predict if a person presently has contracted COVID simply by training models to the symptoms that person has, instead of needing a PCR test. This project drew its datasets from the Israeli Ministry of Health and the West Virginia University Hospital which published the results of all tests and symptom surveys on individuals within a time frame. Eliminating PCR testing would reduce diagnosis times.  As a second goal, a diagnosis of influenza versus COVID would like to be made.

Things I liked about the project:
- Analysis tables for the symptoms were well formatted and clear. 
- Using another dataset to access greater granularity of covariates. 
- Graphic are clear and concise. 

Areas for improvement:
- Explaining why you chose to trim the dataset so that there's a 50-50 split in positive/negative tests. How did you decide which observations to trim?
- Analysis into correlation between values that were NaN and outcomes would have be a valuable addition. Reducing from 7,651,053 to 5,977,416 is significant and I wonder if missing value imputing may have helped models train on all the data. 
- Instead of concluding that overfitting is unlikely, statistical evidence should be added. 
- Discussion as to how 264 patients were on ventilators but only 82 tested positive for flu or COVID. I don't believe it is normal for that many patients to be on ventilators ignoring flu and COVID. 
- You concluded that the model is fair but you rejected the assumption that it was fair. I do not understand.

**Comment by zgk2003** [comment]
Hi there,

Thank you very much for your suggestions!

However, I do think that there are a few points where you misunderstood our report.

- For your first point, I did explain that the reason for the split was that our original dataset is unbalanced. It was covered in lecture that if we have an unbalanced dataset, we should trim it to 50-50 such that the result is unbiased. The process is random (i.e. we randomly dropped some of the negative cases so that the number of positive cases and the number of negative cases in our dataset are equal).
- Secondly, while the drop from 7,651,053 to 5,977,416 is indeed significant numerically, it is not so statistically. This is because we are experiencing a decrease in return as we increase the size of our dataset. For example, a decrease from 200 to 100 is significant, but a decrease from 200,000 to 100,000 is probably not so significant, since most models do not need so much data to train.
- We proposed that underfitting is unlikely before we trained the models, but our accuracies were all measured on validation sets, so the effect of underfitting has been removed in this metric.
- The second to last point is indeed somewhat intuitive. However, when we consider that for a large hospital, there are many other patients with other diseases such as asthma, lung cancer, etc., this is not so abnormal.
- As for your last point, the report reads "we thus fail to reject our null hypothesis." Thus, I believe that you misread our report.

Thank you very much for your suggestions!

**Comment by mcguirecooper** [comment]
If you could direct me to the code for the chi-squared test on fairness, that would be appreciated. I did not see it in any of the github files.

**Comment by zgk2003** [comment]
We actually did that in Stata via Cornell's Apps on Demand since I am learning Stata in my other course this semester and I was trying to get some practice with that. The relevant files are on the virtual machine and thus not pushed to this Github repo.
